Source/gui/SNTBinaryMessageWindowView.swift

```diff
@@ -0,0 +1,313 @@
+/// Copyright 2024 North Pole Security, Inc.
+///
+/// Licensed under the Apache License, Version 2.0 (the "License");
+/// you may not use this file except in compliance with the License.
+/// You may obtain a copy of the License at
+///
+///     https://www.apache.org/licenses/LICENSE-2.0
+///
+/// Unless required by applicable law or agreed to in writing, software
+/// distributed under the License is distributed on an "AS IS" BASIS,
+/// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+/// See the License for the specific language governing permissions and
+/// limitations under the License.
+
+import SwiftUI
+
+import santa_common_CertificateHelpers
+import santa_common_SNTBlockMessage
+import santa_common_SNTConfigurator
+import santa_common_SNTStoredEvent
+
+let MAX_OUTER_VIEW_WIDTH = 560.0
+let MAX_OUTER_VIEW_HEIGHT = 340.0
+let MAX_BUTTON_AREA_WIDTH = 300.0
+
+@objc public class SNTBinaryMessageWindowViewFactory : NSObject {
+  @objc public static func createWith(window: NSWindow,
+                                      event: SNTStoredEvent,
+                                      customMsg: NSString?,
+                                      customURL: NSString?,
+                                      uiStateCallback: ((Bool) -> Void)?) -> NSViewController {
+    return NSHostingController(rootView:SNTBinaryMessageWindowView(
+      window:window,
+      event:event,
+      customMsg:customMsg,
+      customURL:customURL,
+      uiStateCallback:uiStateCallback).frame(
+        minWidth:MAX_OUTER_VIEW_WIDTH,
+        minHeight:MAX_OUTER_VIEW_HEIGHT).fixedSize())
+  }
+}
+
+struct SNTBinaryMessageEventExpandedView: View {
+  let e: SNTStoredEvent?
+
+  @Environment(\.presentationMode) var presentationMode
+
+  func addLabel(@ViewBuilder closure: () -> some View) -> some View {
+    HStack(spacing:5.0) {
+      VStack(alignment:.leading, spacing:2.0) {
+        closure()
+      }.frame(alignment:.leading)
+      Spacer()
+    }.frame(width:MAX_OUTER_VIEW_WIDTH - 60).fixedSize()
+  }
+
+  var body: some View {
+    HStack(spacing: 20.0) {
+      VStack(spacing: 20.0) {
+        Spacer()
+
+        addLabel {
+          Text("Path").bold().font(Font.system(size:12.0))
+          Text(e?.filePath ?? "unknown").textSelection(.enabled)
+        }
+
+        Divider()
+
+        if let signingID = e?.signingID {
+          addLabel {
+            Text("Signing ID").bold().font(Font.system(size:12.0))
+            Text(signingID).font(Font.system(size:12.0).monospaced()).textSelection(.enabled)
+          }
+        }
+
+        Divider()
```

Comment:
If there is no SID, two dividers will be added. I don't think that is what we want?

Maybe move `Divider()` into the if block?

Reply:
Done, and the same for CDHash